src/OICodeGen.cpp

```diff
@@ -986,6 +989,99 @@ std::string OICodeGen::typeToName(drgn_type *type) {
   return typeName;
 }
 
+bool OICodeGen::recordChildren(drgn_type *type) {
+  drgn_type_template_parameter *parents = drgn_type_parents(type);
+
+  for (size_t i = 0; i < drgn_type_num_parents(type); i++) {
+    struct drgn_qualified_type t {};
+
+    if (auto *err = drgn_template_parameter_type(&parents[i], &t)) {
```

Comment:
style: I think the new style is to use (explicitly checking the nullptr):
```
if (auto *err = drgn_type_iterator_create(prog, &typesIterator); err != nullptr) {
...
}
```

Though there's nothing enforcing it yet. There are a few instances of this.

Reply:
Yes, I prefer this style and used that in code cleanup for open sourcing.